packages/astro/src/runtime/server/index.ts

```diff
@@ -766,17 +790,21 @@ export async function renderPage(
 				start(controller) {
 					async function read() {
 						let i = 0;
-						for await (const chunk of iterable) {
-							let html = chunk.toString();
-							if (i === 0) {
-								if (!/<!doctype html/i.test(html)) {
-									controller.enqueue(encoder.encode('<!DOCTYPE html>\n'));
+						try {
+							for await (const chunk of iterable) {
```

Comment:
Nit: I noticed we're using `for await` all over the place in this file, but there's nothing async happening in any loop. Any reason for that?

Reply:
This diff is just from moving this into a `try/catch`, but AFAICT `for await` is correct because AstroComponents are `AsyncIterable`. `for await` signals that we should wait for each chunk.

Reply:
Right, these chunks are async because of streaming.